CMakeLists.txt

```diff
@@ -1,11 +1,38 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 3.15.0)
 
+ENABLE_LANGUAGE(Fortran)
 PROJECT(MultiphaseStokes LANGUAGES C CXX Fortran)
 FIND_PACKAGE(IBAMR REQUIRED)
 
 SET(CXX_SRC
+    VCTwoFluidStaggeredStokesOperator.cpp
+    VCTwoFluidStaggeredStokesBoxRelaxationFACOperator.cpp
     )
-ADD_EXECUTABLE(main2d main.cpp)
-TARGET_SOURCES(main2d PRIVATE ${CXX_SRC})
-TARGET_LINK_LIBRARIES(main2d IBAMR::IBAMR2d)
-CONFIGURE_FILE(input2d ${CMAKE_CURRENT_BINARY_DIR}/input2d COPYONLY)
+
+SET(F_SRC
+    RBGS.f
+    )
+
+ADD_EXECUTABLE(apply2d apply.cpp)
+TARGET_SOURCES(apply2d PRIVATE ${CXX_SRC})
+TARGET_LINK_LIBRARIES(apply2d IBAMR::IBAMR2d)
+
+ADD_EXECUTABLE(solver2d solver.cpp)
+TARGET_SOURCES(solver2d PRIVATE ${CXX_SRC})
+TARGET_LINK_LIBRARIES(solver2d IBAMR::IBAMR2d)
+
+ADD_EXECUTABLE(smooth2d smooth.cpp)
+TARGET_SOURCES(smooth2d PRIVATE ${CXX_SRC} ${F_SRC})
+TARGET_LINK_LIBRARIES(smooth2d IBAMR::IBAMR2d)
+
+ADD_EXECUTABLE(rbgs_smooth2d smooth.cpp)
+TARGET_SOURCES(rbgs_smooth2d PRIVATE ${CXX_SRC} ${F_SRC})
+TARGET_LINK_LIBRARIES(rbgs_smooth2d IBAMR::IBAMR2d)
+
+ADD_EXECUTABLE(fluid_multigrid multigrid.cpp)
+TARGET_SOURCES(fluid_multigrid PRIVATE ${CXX_SRC} ${F_SRC})
+TARGET_LINK_LIBRARIES(fluid_multigrid IBAMR::IBAMR2d)
+
+CONFIGURE_FILE(input2d_five ${CMAKE_CURRENT_BINARY_DIR}/tests/fluid_multigrid/input2d_five COPYONLY)
```

Comment:
Why is this being copied to the `tests/fluid_multigrid` directory?

Reply:
I copied it there to indicate which input file I was using to test the preconditioner, but it isn't necessary to do.

Reply:
I'd suggest not copying any input files. If edit an input file in the copied directory then rerun cmake, it will overwrite the input file. You could consider doing a softlink.

Also, renaming the input files for their purpose (`input_solve`, `input_var_thn`, etc.) instead of `input_one`, `input_two`, etc. seems like a good idea.

Reply:
That makes sense.  I will make those changes.